metacat-metadata-mysql/src/main/java/com/netflix/metacat/metadata/mysql/MySqlLookupService.java

```diff
@@ -244,20 +206,14 @@ private Lookup findOrCreateLookupByName(final String name) throws SQLException {
      * @return returns the lookup with the given name.
      */
     @Override
-    public Lookup addValues(final String name, final Set<String> values) {
+    public Lookup addValues(final String name, final Set<String> values, final boolean includeValues) {
         try {
-            final Lookup lookup = findOrCreateLookupByName(name);
-
-            final Set<String> inserts;
-            final Set<String> lookupValues = lookup.getValues();
-            if (lookupValues == null || lookupValues.isEmpty()) {
-                inserts = values;
-                lookup.setValues(values);
-            } else {
-                inserts = Sets.difference(values, lookupValues);
+            final Lookup lookup = findOrCreateLookupByName(name, includeValues);
+            if (!values.isEmpty()) {
+                insertLookupValuesIfNotExist(lookup.getId(), values);
             }
-            if (!inserts.isEmpty()) {
-                insertLookupValues(lookup.getId(), inserts);
+            if (includeValues) {
```

Comment:
the previous code didn't even add back the new tags to the return result, but based on usage it does not make a difference either way

Reply:
yep, you are right but I want to make it consistent.